plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -343,6 +344,27 @@ public boolean warningExists(Plugin plugin) {
         return !securityWarnings.isEmpty();
     }
 
+    /**
+     * Checks that required Jenkins version of all plugins to be downloaded is less than the Jenkins version in the
+     * user specified Jenkins war file
+     *
+     * @param pluginsToBeDownloaded
+     */
+    public void checkVersionCompatibility(List<Plugin> pluginsToBeDownloaded) {
+        if (jenkinsVersion != null && !StringUtils.isEmpty(jenkinsVersion.toString())) {
+            for (Plugin p : pluginsToBeDownloaded) {
+                if (p.getJenkinsVersion() != null) {
+                    if (p.getJenkinsVersion().compareTo(jenkinsVersion) > 0) {
+                        System.out.println(
```

Comment:
I would either fail or not add the check I think, how likely is that people would see this is in the logs?

newer jenkins core normally means newer apis are consumed so it wouldn't be safe to proceed anyway?

Reply:
Why did you first say you might not add the check at all?